crypto/fipsmodule/ecdh/ecdh.c

```diff
@@ -79,26 +79,43 @@
 
 int ECDH_compute_shared_secret(uint8_t *buf, size_t *buflen, const EC_POINT *pub_key,
                                const EC_KEY *priv_key) {
+  int ret = 0;
+  EC_KEY *key_pub_key = NULL;
   if (priv_key->priv_key == NULL) {
     OPENSSL_PUT_ERROR(ECDH, ECDH_R_NO_PRIVATE_VALUE);
-    return 0;
+    goto end;
   }
   const EC_SCALAR *const priv = &priv_key->priv_key->scalar;
   const EC_GROUP *const group = EC_KEY_get0_group(priv_key);
   if (EC_GROUP_cmp(group, pub_key->group, NULL) != 0) {
     OPENSSL_PUT_ERROR(EC, EC_R_INCOMPATIBLE_OBJECTS);
-    return 0;
+    goto end;
+  }
+
+  // |EC_KEY_check_fips| is not an expensive operation on an external
+  // public key.
+  key_pub_key = EC_KEY_new();
```

Comment:
Should you check for null here before using key_pub_key?

Reply:
Thanks. Done.